packages/commerce-sdk-react/src/types.ts

```diff
@@ -0,0 +1,24 @@
+interface CommonHookResponse {
+    error: Error | undefined
+    isLoading: boolean
+}
+
+// These are the common params for all query hooks
+// it allows user to override configs for specific query
+interface QueryParams {
+    siteId?: string
+    locale?: string
+    currency?: string
+    organizationId?: string
+    shortCode?: string
+}
+
+interface QueryResponse<data> extends CommonHookResponse {
+    data: data
+}
+
+interface ActionResponse<execute> extends CommonHookResponse {
+    execute: execute
```

Comment:
I'm still a newbie when it comes to typing stuffs. If we know that `execute` would be a function, does it need to be a generic? Or the type can be just `Function` like this?

```suggestion
interface ActionResponse extends CommonHookResponse {
    execute: Function
```

Reply:
Best practice is to use type signatures, rather than `Function`. `Function` is equivalent to `(...any[]) => any`. We should try to be as specific as possible, something like `(params: Record<string, unknown>, body?: unknown) => Promise<unknown>` or whatever we have for the implementations at this point.